### DISCUSSION TITLE
Running mockoon in docker/kubernetes

### DISCUSSION THREAD

**Discussion by davidkiss96** [mockoon/mockoon#771] [Help]
Hello Community.

We want to run mockoon in a dockerized environment,
I would need some insight/advice for the following points:

1. how to load json data into the cli

2. can a dynamic port set? (cant set a direct port in kubernetes, so we need the possibility override port dynamically)

3. is there a way to have a generated json documentation? not sure, maybe based on rules, description of response etc, so if a new dev joins the team can check what the JSON file is about, what it does, what is it responsible for without loading the file into code editor or into mockoon.

Thanks in advance.

**Comment by 255kb** [comment]
Ho, 

I don't if you had a look at our [documentation](https://github.com/mockoon/mockoon/tree/main/packages/cli#docker). We provide a Docker image which embarks Mockoon CLI and to which you provide most parameters at runtime. 

You have first to identify where you Mockoon's data file is located, put it somewhere where your container can access it, and then run a similar command: `docker run -d --mount type=bind,source=/home/your-data-file.json,target=/data,readonly -p 3000:3000 mockoon/cli:latest --data data --port 3000`
You will be able to use the same flags that are available for the [`start` command](https://github.com/mockoon/mockoon/tree/main/packages/cli#mockoon-cli-start) and dynamically set the port at runtime. 

Concerning your last question, there is no such feature right now. The mock file is supposed to be explored using the UI. You can export to an OpenAPI spec file, but you have to be aware that there is [no 1:1 equivalent](https://mockoon.com/docs/latest/openapi/openapi-specification-compatibility/) between a Mockoon file and an OpenAPI spec. 

Let me know if you have questions.
